src/main/java/Parser.java

```diff
@@ -0,0 +1,92 @@
+public final class Parser {
+    /**
+     * Parse the <code>commandOptions</code> argument for {@link ActionManager#markUnmarkTask(String, String) markUnmarkTask} method
+     * @param args <code>commandOptions</code> received from upstream
+     * @return Task number (starts from 1)
+     * @throws SuBotException If invalid task number
+     */
+    static int parseMarkUnmark(String args) throws SuBotException {
+        if (args.isBlank()) {
+            throw new SuBotException("Task number cannot be blank");
+        }
+        try {
+            int taskIndex = Integer.parseUnsignedInt(args.trim()) - 1; //Index starts from 0
+            if (taskIndex < 0) {
+                throw new SuBotException("Task number must be a positive integer");
+            } else { return taskIndex; }
```

Comment:
Follow the layout convention for if-else statements, format it using Egyptian style